### DISCUSSION TITLE
hssm is not installable on windows (cpu only)

### DISCUSSION THREAD

**Discussion by yufeng-tian** [lnccbrown/HSSM#470] [General]
I follow the official installing steps on windows (cpu only). When run

(hssm) C:\Users\tianl>conda install -c conda-forge hssm

It reported that
Could not solve for environment specs
The following package could not be installed
└─ hssm is not installable because it requires
└─ numpyro >=0.14.0,<0.15.0 , which requires
└─ jaxlib >=0.4.14 , which does not exist (perhaps a missing channel).

However, it was showed that jaxlib-0.4.30 had been installed (see screenshot)

![85f260dd-a98f-4b4b-bf02-c0fe81164d71](https://github.com/lnccbrown/HSSM/assets/54008634/eba30384-7ff4-4a32-b52e-12e972348909)

I have tried Python 3.11 and 3.10, I've also tried to create a new environment through the channel of conda-forge (conda create -c conda-forge -n hssm python=3.11). Unfortunately, they did not work.

Do you know how I can solve this problem?

thank you so much

**Comment by naziajassim** [comment]
Just adding that I've run into the exact same issue with hssm installation on windows. I've been using hssm for the past couple months and  decided to update to the latest version after running into new errors while doing the tutorial. Now I can't seem to install it again. Would appreciate any suggestions. Thanks!

**Reply by Hellobamboobamboo** [reply]
I have exact same issue too. All new required versions, still report old.

**Comment by digicosmos86** [comment]
Thanks for letting us know. The problem seems to be that `hssm` has a dependency on conda-forge of `numpyro`, which specified `jaxlib` as a dependency, which is not available for windows on conda-forge, resulting in the failure. It seems that reverting back to the old way of installing hssm would solve this problem. You just need to install pymc through conda-forge first, and then install hssm through pip:

```sh
conda install -c conda-forge pymc=5.14
pip install hssm
```

This would get around of installing numpyro through conda-forge. We really only need pymc through conda-forge.

Please let me know if it works

**Reply by yufeng-tian** [reply]
Hi Paul, 

Thank you so much. This method works for me! HSSM has been successfully installed. I tried the testing model just now and everything seemed to work properly.

One point is that if the system reports error regarding jupyter and ipywidgets, just follow the suggestion to update and install related packages: 
Warning: IProgress not found. Please update jupyter and ipywidgets. See [https://ipywidgets.readthedocs.io/en/stable/user_install.html](url)
 from .autonotebook import tqdm as notebook_tqdm

**Reply by naziajassim** [reply]
Thank you Paul! The installation works fine after using your suggestion and after updating jupyter and ipywidgets as suggested by Yufeng. Cheers both.

**Reply by jry4n** [reply]
thank you, Paul!  This should really be noted on the installation page, especially when it says:

> Important Update: From HSSM 0.2.2, conda-forge is the official way of installing HSSM. This will also install other libraries such as libblas that PyMC requires to run properly.

yet, pip seems to be the only way to currently install this on windows

**Comment by akraskovSMDB** [comment]
Hi, did not help me.
pymc was installed but hssm installation failed (see below)
What python version shall I specify when creating the environment? Could this be an explanation for the problem? 

(hssm) C:\Temp>pip install hssm
ERROR: Ignored the following versions that require a different python version: 0.1.0 Requires-Python >=3.9,<3.12; 0.1.1 Requires-Python >=3.9,<3.12; 0.1.2 Requires-Python >=3.9,<3.12; 0.1.3 Requires-Python >=3.9,<3.12; 0.1.4 Requires-Python >=3.9,<3.12; 0.1.4.post3 Requires-Python >=3.9,<3.12; 0.1.5 Requires-Python >=3.9,<3.12; 0.2.0 Requires-Python >=3.10,<3.12; 0.2.0b1 Requires-Python >=3.10,<3.12; 0.2.1 Requires-Python <3.12,>=3.10; 0.2.2 Requires-Python <3.12,>=3.10; 0.2.3 Requires-Python <3.12,>=3.10
ERROR: Could not find a version that satisfies the requirement hssm (from versions: none)
ERROR: No matching distribution found for hssm

**Reply by digicosmos86** [reply]
Hi,

HSSM is compatible with Python 3.10 and 3.11 at the moment. Which Python version do you have in your environment?

**Reply by akraskovSMDB** [reply]
Had to reinstall anaconda, but now managed to install hssm in env with Python 3.10, thank you

**Comment by Hellobamboobamboo** [comment]
Do we have to install in cmd and conda prompt both? twice?

**Reply by digicosmos86** [reply]
Can you provide more details on what steps you followed to create your virtual environment and what is now installed in your environment? Any error messages?

**Comment by juneeybug** [comment]
@digicosmos86 In Windows CPU, if conda install -c conda-forge pymc=5.14 does not work (keeps failing to install) , how can we go around to install the latest hssm?

**Reply by digicosmos86** [reply]
Hi @juneeybug,

What was the issue in installing PyMC via conda?

There are a few things that you can try:
1. Install the latest version of PyMC: `conda install -c conda-forge pymc` and then install HSSM. We just released a new version of HSSM that is compatible with the latest version of pymc
2. Install HSSM via pip directly: `pip install hssm`. You might potentially run into BLAS issues. If you get warnings like `WARNING (pytensor.tensor.blas): Using NumPy C-API based implementation for BLAS functions.`, let us know and we can try to resolve that
3. Install HSSM via conda in WSL: this might potentially be the easiest way if you don't mind working in WSL
